### ISSUE TITLE
Unable to run either model due to SciPy sparse matrix errors

### ISSUE THREAD

**Issue by rrajasek95** [Zziwei/Fairness-in-Cold-Start-Recommendation#1]
Hello, I was trying to run the `Gen` and `Scale` variants of the model, but I kept getting the following error due to the evaluation portion of the code:
```
  File "main.py", line 239, in <module>
    main()
  File "main.py", line 126, in main
    recall_k=recall_at, eval_data=cold_test_eval, R=R)
  File "/<path>/Fairness-in-Cold-Start-Recommendation/Scale/utils.py", line 98, in batch_eval_recall
    z = y.multiply(x)
  File "/<path>/anaconda2/envs/fairness_recs_tf1/lib/python3.6/site-packages/scipy/sparse/base.py", line 336, in multiply
    return self.tocsr().multiply(other)
  File "/home/athtrch/anaconda2/envs/fairness_recs_tf1/lib/python3.6/site-packages/scipy/sparse/compressed.py", line 373, in multiply
    other = self.__class__(other)
  File "/<path>/anaconda2/envs/fairness_recs_tf1/lib/python3.6/site-packages/scipy/sparse/compressed.py", line 34, in __init__
    arg1 = arg1.asformat(self.format)
  File "/<path>/anaconda2/envs/fairness_recs_tf1/lib/python3.6/site-packages/scipy/sparse/base.py", line 322, in asformat
    return convert_method(copy=copy)
  File "/<path>/anaconda2/envs/fairness_recs_tf1/lib/python3.6/site-packages/scipy/sparse/lil.py", line 462, in tocsr
    _csparsetools.lil_get_lengths(self.rows, indptr[1:])
  File "_csparsetools.pyx", line 109, in scipy.sparse._csparsetools.lil_get_lengths
```

I presume the error occurs due to the following lines (90-96) in `utils.py` under batch_eval_recall:
```
preds_k = preds_all[:, :at_k]
at_k = preds_k.shape[1]
y = eval_data.R_test_inf

x = scipy.sparse.lil_matrix(y.shape)
x.rows = preds_k
x.data = np.ones_like(preds_k)
```

Where data from a smaller matrix is assigned to a bigger one. This causes the multiply to fail since perhaps the data is stored in a different layout under LIL matrices. Would you be able to provide the SciPy version for your code since I'm having trouble getting it to work? 

Thank you.

**Comment by Zziwei** [comment]
Hi,

I think it might be the issue with the scipy version. I used scipy 1.4.1, and I tried to update it to 1.7.0 and I faced the same problem as yours. 

I will try to fix it so that it can work for scipy 1.7.0. But right now, you can try to degrade it to 1.4.1 to run the code.

**Comment by Zziwei** [comment]
I updated the code to avoid using lil_matrix. Now it works for scipy 1.7.0
